db/db_impl/db_impl.cc

```diff
@@ -4789,6 +4826,44 @@ Status DBImpl::VerifyChecksum(const ReadOptions& read_options) {
   return s;
 }
 
+Status DBImpl::VerifySstFileChecksum(const FileMetaData& fmeta,
+                                     const std::string& fname,
+                                     const std::string& file_checksum_func_name,
+                                     const ReadOptions& read_options) {
+  Status s;
+  if (fmeta.file_checksum == kUnknownFileChecksum) {
+    return s;
+  }
+  if (fmeta.file_checksum_func_name != file_checksum_func_name) {
+    std::ostringstream oss;
+    oss << fname
+        << " checksum func name does not "
+           "match ("
+        << fmeta.file_checksum_func_name << "vs. " << file_checksum_func_name
+        << ")";
+    s = Status::InvalidArgument(oss.str());
+    return s;
+  }
+  std::string file_checksum;
+  std::string func_name;
+  s = ROCKSDB_NAMESPACE::GenerateOneFileChecksum(
```

Comment:
I will quickly create a PR to fix `GenerateOneFileChecksum`.  Zitan introduced the `requested_checksum_func_name` in `file_checksum_contex`, but this function is not updated accordingly......

Reply:
@zhichao-cao, so after your fix, `GenerateOneFileChecksum()` will take `func_name` as input.
It will be used to set `file_checksum_contex.requested_checksum_func_name` which will be used to request a function from factory. If no function is returned or the returned func has a different name, then there is an error. Is this right?

Reply:
In GenerateOneFileChecksum, we pass in the `requested_checksum_func_name` and it will get the checksum together with the `generator_checksum_func_name`. GenerateOneFileChecksum does not compare and verify if `requested_checksum_func_name == generator_checksum_func_name`. I think caller will do the verification if they want, and decide to use the checksum or not.